02-express-tutorial/controlers/people.js

```diff
@@ -0,0 +1,59 @@
+let { people } = require("../data");
+
+const getPeople = (req, res) => {   
+    res.status(200).json({ success: true, data: people });
+}
+const createPerson = (req, res) => {
+    const { name } = req.body;
+    if(!name){
+        return res
+            .status(400)
+            .json({ success: false, msg: "Please provide name value" });
+    }
+    res.status(201).json({ success: true, person: name }); 
+}
+const createPersonPostman =  (req, res) => {
+    const { name } = req.body;
+    if(!name){
+        return res
+            .status(400)
+            .json({ success: false, msg: "Please provide name value" });
+    }
+    res.status(201).json({ success: true, data: [...people, name] }); 
+}
+const updatePerson = (req, res) => {
+    const { id } = req.params;
+    const { name } = req.body;
+    const person = people.find((person) => person.id === Number(id));
+    if(!person){
+        return res
+            .status(404)
+            .json({ success: false, msg: `no person with id ${id}` });
+    }
+    const newPeople = people.map((person) => {
+    if(person.id === Number(id)){
+        person.name = name;
+    }
+    return person;
+    });
+    res.status(200).json({ success: true, data: newPeople });
```

Comment:
Requires change:
This can be simplified to using `Array.find`:
```js
const personId = parseInt(req.params.id, 10);

const person = people.find(person => person.id === personId);

if (!person) {
  return res.status(404).json({ success: false, msg: `no person with id ${personId}` });
} 

person.name = req.body.name;

res.status(200).json({ success: true, data: person });
```

Reply:
Wow. That's really awesome!!! Thank you for your suggestion!